test/functional/ui/incsubstitution_spec.lua

```diff
@@ -0,0 +1,1214 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear = helpers.clear
+local curbufmeths = helpers.curbufmeths
+local eq = helpers.eq
+local eval = helpers.eval
+local execute = helpers.execute
+local expect = helpers.expect
+local feed = helpers.feed
+local insert = helpers.insert
+local meths = helpers.meths
+local neq = helpers.neq
+local source = helpers.source
+
+local default_text = [[
+  Inc substitution on
+  two lines
+]]
+
+local function common_setup(screen, incsub, text)
+  if screen then
+    execute("syntax on")
+    execute("set nohlsearch")
+    execute("hi IncSubstitute guifg=red guibg=yellow")
+    screen:attach()
+    screen:set_default_attr_ignore( {{bold=true, foreground=Screen.colors.Blue}} )
+    screen:set_default_attr_ids({
+      [1]  = {foreground = Screen.colors.Fuchsia},
+      [2]  = {foreground = Screen.colors.Brown, bold = true},
+      [3]  = {foreground = Screen.colors.SlateBlue},
+      [4]  = {bold = true, foreground = Screen.colors.SlateBlue},
+      [5]  = {foreground = Screen.colors.DarkCyan},
+      [6]  = {bold = true},
+      [7]  = {underline = true, bold = true, foreground = Screen.colors.SlateBlue},
+      [8]  = {foreground = Screen.colors.Slateblue, underline = true},
+      [9]  = {background = Screen.colors.Yellow},
+      [10] = {reverse = true},
+      [11] = {reverse = true, bold=true},
+      [12] = {foreground = Screen.colors.Red, background = Screen.colors.Yellow},
+      [13] = {bold = true, foreground = Screen.colors.SeaGreen},
+      [14] = {foreground = Screen.colors.White, background = Screen.colors.Red},
+    })
+  end
+
+  if incsub then
```

Comment:
is this `if` needed? can't we make sure that incsub always is the empty string if it should be off.